src/app/content/components/Page/PageComponent.tsx

```diff
@@ -47,6 +47,19 @@ export default class PageComponent extends Component<PagePropTypes> {
 
       transformContent(parsedContent, parsedContent.body, this.props.intl);
 
+      if (this.props.lockNavigation) {
+        const links = parsedContent.body.querySelectorAll('a[href]');
+
+        for (const link of links) {
+          const {origin, pathname} = new URL(link.getAttribute('href'), assertWindow().location.href);
```

Comment:
i think you can use this utility here https://github.com/openstax/rex-web/blob/ecb62adf6d2bad690953b5b31a3665d5f5058e23/src/app/content/utils/urlUtils.ts#L104
```suggestion
          const pathname = fromRelativeUrl(link.getAttribute('href'), this.props.currentPath);
```